httptransport/client/indexer.go

```diff
@@ -13,43 +12,50 @@ import (
 	clairerror "github.com/quay/clair/v4/clair-error"
 	"github.com/quay/clair/v4/httptransport"
 	"github.com/quay/clair/v4/indexer"
+	"github.com/quay/clair/v4/internal/codec"
 )
 
 var _ indexer.Service = (*HTTP)(nil)
 
 func (s *HTTP) AffectedManifests(ctx context.Context, v []claircore.Vulnerability) (*claircore.AffectedManifests, error) {
-	var affected claircore.AffectedManifests
-	buf := bytes.NewBuffer([]byte{})
-	err := json.NewEncoder(buf).Encode(struct {
-		V []claircore.Vulnerability `json:"vulnerabilities"`
-	}{
-		v,
-	})
-	if err != nil {
-		return nil, &clairerror.ErrBadVulnerabilities{err}
-	}
-
 	u, err := s.addr.Parse(httptransport.AffectedManifestAPIPath)
 	if err != nil {
 		return nil, fmt.Errorf("failed to parse api address: %v", err)
 	}
-	req, err := http.NewRequestWithContext(ctx, "POST", u.String(), buf)
+	rd := codec.JSONReader(struct {
+		V []claircore.Vulnerability `json:"vulnerabilities"`
+	}{
+		v,
+	})
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost, u.String(), rd)
 	if err != nil {
 		return nil, fmt.Errorf("failed to create request: %v", err)
 	}
+	req.Header.Set("content-type", `application/json`)
 	resp, err := s.c.Do(req)
 	if err != nil {
 		return nil, err
 	}
 	defer resp.Body.Close()
 	if resp.StatusCode != http.StatusOK {
-		return nil, &clairerror.ErrRequestFail{Code: resp.StatusCode, Status: resp.Status}
-	}
-	err = json.NewDecoder(resp.Body).Decode(&affected)
-	if err != nil {
-		return nil, &clairerror.ErrBadAffectedManifests{err}
+		return nil, &clairerror.ErrRequestFail{
```

Comment:
Why the change in error type?

Reply:
I don't know what you mean, it's still `ErrRequestFail`.

Reply:
Misread that